docs/side_quests/ide_features.md

```diff
@@ -0,0 +1,654 @@
+4. Now you can see both files side by side
+
+![Split editor](img/split_editor.png)
+
+This is invaluable when:
+
+- Checking module interfaces while writing workflow calls
+- Comparing similar processes across different modules
+- Debugging data flow between workflow and modules
+
+### 5.3. Project-wide Search
+
+Sometimes you need to find where specific patterns are used across your entire project. Press **Ctrl+Shift+F** (or **Cmd+Shift+F**) to open the search panel.
+
+Try searching for `publishDir` across the workspace:
+
+![Project search](img/project_search.png)
+
+This shows you every file that uses publish directories, helping you:
+
+- Understand output organization patterns
+- Find examples of specific directives
+- Ensure consistency across modules
+
+### Takeaway
+
+You can manage complex multi-file projects using quick file navigation, split editors, and project-wide search to work efficiently across workflows and modules.
+
+### What's next?
+
+Learn how code formatting and maintenance features keep your workflows organized and readable.
+
+---
+
+## 6. Code Formatting and Maintenance
+
+Good code formatting isn't just about aesthetics - it makes complex workflows easier to understand and maintain.
+
+### 6.1. Automatic Formatting in Action
+
+Open `basic_workflow.nf` and deliberately mess up the formatting:
+
+- Remove some indentation
+- Add extra spaces in random places
+- Break some lines awkwardly
+
+Now press **Shift+Alt+F** (or **Shift+Option+F**) to auto-format:
+
+VS Code immediately:
+
+- Fixes indentation to show process structure clearly
+- Aligns similar elements consistently
+- Removes unnecessary whitespace
+- Maintains readable line breaks
+
+### 6.2. Code Organization Features
+
+#### Quick Commenting
+
+Select a block of code in your workflow and press **Ctrl+/** (or **Cmd+/**) to comment it out:
+
+```groovy
+// workflow {
+//     ch_input = Channel.fromPath(params.input)
+//         .splitCsv(header: true)
+//         .map { row -> [row.sample_id, file(row.fastq_path)] }
+//
+//     FASTQC(ch_input)
+// }
+```
+
+This is perfect for:
+
+- Temporarily disabling parts of workflows during development
+- Adding explanatory comments to complex channel operations
+- Documenting workflow sections
+
+Use **Ctrl+/** (or **Cmd+/**) again to uncomment the code.
+
+#### Code Folding for Overview
+
+In `complex_workflow.nf`, notice the small arrows next to process definitions. Click them to fold (collapse) processes:
+
+![Code folding](img/code_folding.png)
+
+This gives you a high-level overview of your workflow structure without getting lost in implementation details.
+
+#### Bracket Matching
+
+Place your cursor next to any `{` or `}` bracket and VS Code highlights the matching bracket. Use **Ctrl+Shift+\\** (or **Cmd+Shift+\\**) to jump between matching brackets.
+
+This is crucial for:
+
+- Understanding process boundaries
+- Finding missing or extra brackets
+- Navigating nested workflow structures
+
```

Comment:
multi-line selection might be another nice one to add (opt+cmd+up/down arrow). 
multi-line highlighting and indentin/outdenting blocks of code

I use it all the time also in trainings.